pkg/kubelet/cm/deviceplugin/endpoint_handler.go

```diff
@@ -0,0 +1,224 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package deviceplugin
+
+import (
+	"fmt"
+	"sync"
+	"time"
+
+	pluginapi "k8s.io/kubernetes/pkg/kubelet/apis/deviceplugin/v1alpha"
+)
+
+type endpointHandler interface {
+	NewEndpoint(resourceName, socketPath string) error
+
+	Devices() map[string][]pluginapi.Device
+	Endpoint(resourceName string) (endpoint, bool)
+
+	Stop() error
+
+	Store() endpointStore
+	SetStore(endpointStore)
+
+	Callback() monitorCallback
+	SetCallback(f monitorCallback)
+}
+
+type endpointStore interface {
+	Endpoint(resourceName string) (endpoint, bool)
+	SynchronizedEndpoint(resourceName string) (*synchronizedEndpoint, bool)
+
+	SwapEndpoint(e endpoint) (endpoint, bool)
+	DeleteEndpoint(resourceName string) error
+
+	Range(func(string, *synchronizedEndpoint))
```

Comment:
This was answered in another comment (hidden now):
- This is here here because this is needed to access the internals of the objects during tests.
- Range() is a bit similar as it allows for simplification internally and for testing